packages/grpc-js-core/src/index.ts

```diff
@@ -5,36 +5,86 @@ import { Client } from './client';
 import { Status} from './constants';
 import { makeClientConstructor, loadPackageDefinition } from './make-client';
 import { Metadata } from './metadata';
+import { IncomingHttpHeaders } from 'http';
 
-const notImplementedFn = () => { throw new Error('Not implemented'); };
+export interface OAuth2Client {
+  getRequestMetadata: (url: string) => Promise<{ Authorization: string }>;
```

Comment:
Unfortunately, in order to match the API of the existing library, it has to accept the type of `OAuth2Client` that would be created by `google-auth-library` version 0.9.x. Maybe in the future it could be more flexible and accept both, but for now we should focus on matching the existing API.

Reply:
Done